components/accelerator/nvidia/temperature/component.go

```diff
@@ -88,25 +90,29 @@ func (c *component) Close() error {
 // run this periodically
 func (c *component) CheckOnce() {
 	log.Logger.Infow("checking temperature")
-	d := Data{
+
+	d := collectData(c.nvmlInstance, c.getTemperatureFunc)
+
+	c.lastMu.Lock()
+	c.lastData = d
+	c.lastMu.Unlock()
+}
+
+func collectData(nvmlInstance nvidianvml.InstanceV2, getTemperatureFunc func(uuid string, dev device.Device) (nvidianvml.Temperature, error)) *Data {
```

Comment:
Probably we don't need to pass the getTemperatureFunc here.. If we would like to mock it, we would mock the InstanceV2 interface, just override the GetTemeprature method and if there is no corresponding method, we would add it to support mock tests.

Reply:
Anyway, we should focus on improving the InstanceV2 which should support all NVIDIA GPU related oprations needed by us. So that we can inject failure for E2E tests...

Reply:
@eahydra  Yeah, only reason we do this, is not to regress on the unit test coverage. We will certainly clean up the NVML interface